ReactiveCocoa/Swift/Property.swift

```diff
@@ -134,50 +132,21 @@ FIXME: DISABLED DUE TO COMPILER BUG
 /// The binding will automatically terminate when the property is deinitialized,
 /// or when the created signal sends a `Completed` event.
 public func <~ <T>(property: MutableProperty<T>, producer: SignalProducer<T, NoError>) -> Disposable {
-	let disposable = CompositeDisposable()
-	let propertyDisposable = property.producer.start(completed: {
-		disposable.dispose()
-	})
-
-	disposable.addDisposable(propertyDisposable)
-
+	var disposable: Disposable!
+	
 	producer.startWithSignal { signal, signalDisposable in
-		disposable.addDisposable(signalDisposable)
-
-		signal.observe(next: { [weak property] value in
-			property?.value = value
-			return
-		}, completed: {
-			disposable.dispose()
-		})
+		property <~ signal
```

Comment:
The reason is, disposing an _observation_ to a `Signal` is not the same as disposing the work of the `SignalProducer` that created it. Signals can have N observers, so the disposal of any of them shouldn't affect delivery to the others.